### ISSUE TITLE
Lucky to find this repository

### ISSUE THREAD

**Issue by Mankvis** [pidahbus/deep-image-orientation-angle-detection#2]
I have been working on image 360 angle recognition these days, but there has been no progress until I found your repository. Do you have a pytorch version of the implementation? I look forward to replying

**Comment by pidahbus** [comment]
Hi @Mankvis unfortunately no. If you understand tensorflow, then it is very easy to convert the code into pytorch. You just have to write custom loss function that we defined. Rest is pretty straightforward as we used pre-trained model.

**Comment by Mankvis** [comment]
> Hi @Mankvis unfortunately no. If you understand tensorflow, then it is very easy to convert the code into pytorch. You just have to write custom loss function that we defined. Rest is pretty straightforward as we used pre-trained model.

At present, I am using the Resnet50 network. Each picture is saved as a picture every 3 degrees, 360/3=120, then 120 is used as the number of classifications, and the loss uses the conventional cross entropy. Please only need to process the loss here. Is it alright?

**Comment by pidahbus** [comment]
Yes. You have to implement the loss function only. This loss function uses a modified version of L1 loss designed especially for angles. In that case, your problem will be changed from classification to regression. Therefore, you can use this loss function or any other suitable metric to evaluate your model as typical metrics for classification such as accuracy, F1-score will not work here.

**Comment by Mankvis** [comment]
> Yes. You have to implement the loss function only. This loss function uses a modified version of L1 loss designed especially for angles. In that case, your problem will be changed from classification to regression. Therefore, you can use this loss function or any other suitable metric to evaluate your model as typical metrics for classification such as accuracy, F1-score will not work here.

I changed it according to your method, but the loss has not converged. The verification output is all decimals, which is very sad. It is convenient to disclose your leisure time. Can I communicate with you online in your leisure time? If so, what is the best tool to use?

**Comment by Mankvis** [comment]
> Yes. You have to implement the loss function only. This loss function uses a modified version of L1 loss designed especially for angles. In that case, your problem will be changed from classification to regression. Therefore, you can use this loss function or any other suitable metric to evaluate your model as typical metrics for classification such as accuracy, F1-score will not work here.

The loss is around 30, the success rate is 0, and my validation output is
tensor([[59.0678],
         [58.9441],
         [58.9374],
         [58.9447],
         [59.3153],
         [59.0305],
         [59.4223],
         [59.0301],
         [59.0350],
         [58.8756],
         [58.9470],
         [58.8894],
         [58.9488],
         [58.9493],
         [58.9457],
         [58.9411]], device='cuda:0')
This looks weird, I don't know if there is something wrong with my network
